internal/storage/local_chunk_manager.go

```diff
@@ -70,13 +79,17 @@ func (lcm *LocalChunkManager) Reader(filePath string) (FileReader, error) {
 func (lcm *LocalChunkManager) Write(filePath string, content []byte) error {
 	absPath := path.Join(lcm.localPath, filePath)
 	dir := path.Dir(absPath)
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
+	exist, err := lcm.Exist(dir)
+	if err != nil {
+		return err
+	}
+	if !exist {
 		err := os.MkdirAll(dir, os.ModePerm)
```

Comment:
Should we use `os.ModePerm` here? `os.ModePerm` here is an alias of `0o777`.

Reply:
I will leave it here since this is not part of issue~
But we probably need to review local_chunk_manager carefully